Final putnam axiom bm

brando90 opened the pull request:
### 📊 Dataset Summary

The **Putnam-AXIOM benchmark** is a new evaluation suite designed to measure mathematical reasoning capabilities of large language models (LLMs). It consists of:

- **Putnam-AXIOM Original**:  
  522 problems sourced from historical *William Lowell Putnam Mathematical Competitions*, curated to test advanced problem-solving abilities.

- **Putnam-AXIOM Variation**:  
  100 functionally equivalent problems generated through programmatic transformations (e.g., modifying variables, constants, and surface features) to preserve difficulty while mitigating contamination from publicly available datasets.

- **Hugging Face Dataset Repository**:  
  The benchmark is also hosted on Hugging Face for public access and reuse.

---

### 🧪 Tasks

This PR introduces three evaluation tasks:

- `putnam_axiom_original`:  
  The full set of 522 original Putnam problems.

- `putnam_axiom_variations`:  
  The 100 transformed variation problems, designed for robustness testing.

- `putnam_axiom_variations_org`:  
  The original counterparts for each of the 100 variation problems.

---

### 🛠 Notes

- The `utils.py` file is adapted from the `minerva_math` benchmark, which had a similar problem structure.  
- Minor modifications were made to support Putnam-specific formatting and variation tracking logic.

---

### 🔗 Current Paper

- 📄 OpenReview Submission NeurIPS Workshop Paper: [Putnam-AXIOM](https://openreview.net/forum?id=YXnwlZe0yf&noteId=yrsGpHd0Sf)
- 📄 OpenReview Submission ICML Conference Paper: [Putnam-AXIOM](https://openreview.net/group?id=ICML.cc/2025/Conference/Authors&referrer=%5BHomepage%5D(%2F))
- 🤗  [HF data set](https://huggingface.co/Putnam-AXIOM)

emxia18 committed:
yaml and util files for axiom dataset

emxia18 committed:
fix prompting error + errors from inputing prompts individually

emxia18 committed:
modify code to automatically save csv outputs -- directly modified lm eval code

emxia18 committed:
save binary count

emxia18 committed:
modifications to be adaptable with variations

emxia18 committed:
final changes to axiom task

emxia18 committed:
rename axiom to putnam axiom

emxia18 committed:
small modification to how process results is reading in candidates

emxia18 committed:
add imo olympiad and remove axiom folders

emxia18 committed:
finalized version

Commit:
Add Huggingface Dataset

brando90 committed:
new tasks

brando90 committed:
eval on ampere8

emxia18 committed:
revert modification in evaluator and update prompts

emxia18 committed:
Merge branch 'main' of https://github.com/emxia18/lm-evaluation-harness into main

brando90 committed:
fixed the random comma

brando90 committed:
changes

brando90 committed:
push

emxia18 committed:
prompting experiments

emxia18 committed:
Merge branch 'main' of https://github.com/emxia18/lm-evaluation-harness into main

emxia18 committed:
new variations

emxia18 committed:
merge into main

emxia18 committed:
restore back to pulling from hugging face instead of localling

emxia18 committed:
remove unnecessary imports

emxia18 committed:
Merge branch 'main' of https://github.com/EleutherAI/lm-evaluation-harness into main

emxia18 committed:
code cleanup

emxia18 committed:
change logger

emxia18 committed:
switch hugging face datasets

emxia18 committed:
Merge branch 'main' of https://github.com/EleutherAI/lm-evaluation-harness into main

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2946) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2946) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: emxia18<br/>:white_check_mark: StellaAthena<br/>:white_check_mark: brando90<br/>:x: Kai Fronsdal<br/><hr/>**Kai Fronsdal** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2946) it.</sub>

brando90 commented:
this is kai https://github.com/kaifronsdal @kaifronsdal he should sign the CLA soon!

StellaAthena commented:
Congrats on the paper acceptance and thank you for the submission!

Is this code the official implementation for the paper? If not, can you report testing showing its equivalent to the official implementation?

brando90 committed:
Update dataset path to Putnam-AXIOM/putnam-axiom-dataset-ICML-2025-522 and use correct split names

brando90 committed:
Add base putnam_axiom task for simplified usage

brando90 committed:
Fix process_variations to properly filter for variation=1, as the dataset only contains variations with this value

brando90 committed:
Fix formatting issues in putnam_axiom YAML files by ensuring proper newline at the end of the files.

and added putnam_axiom.yaml because otherwise the lm_eval --tasks list wouldn't work.

StellaAthena commented:
Bumping this :)

baberabb reviewed (changes requested):
Hi! just some nits

brando90 commented:
sorry we are behind on this, we will address this! @emxia18 maybe you can assist me partiall with some of it? 

ps: I don't know how to fix @kaifronsdal's name not appear for the signature part.

brando90 commented:
PS: if anyone reads this and wants to use our code, you can meanwhile look at our code here https://github.com/brando90/lm-evaluation-harness and or ping me personally by opening a gitissue in my repo

baberabb commented on lm_eval/tasks/imo_olympiad/utils.py:
```diff
@@ -0,0 +1,320 @@
+import antlr4
+import datasets
+
+from lm_eval.utils import eval_logger
```
```suggestion
eval_logger = logging.getLogger(__name__)
```

baberabb commented on lm_eval/tasks/imo_olympiad/utils.py:
```diff
@@ -0,0 +1,320 @@
+
+from lm_eval.utils import eval_logger
+
+import numpy as np
```
this import can be removed, as it in unused.

baberabb commented on lm_eval/tasks/putnam_axiom/utils.py:
```diff
@@ -0,0 +1,338 @@
+import antlr4
+import datasets
+
+import numpy as np
```
unused. can removed

baberabb commented on lm_eval/tasks/putnam_axiom/utils.py:
```diff
@@ -0,0 +1,338 @@
+import signal
+from typing import Dict, List, Optional
+
+import antlr4
```
unused. can remove

baberabb commented:
> we will do everything here soon! Thank you for your patience and support/feedback 🙏
> 
> Is it possible to ignore or remove the request for Kai? Idk why github is being annoying about it...we can't remove it.

Hi, no problem, we can bypass that :). Mainly some minor formatting issues left. I tried fixing them on my end, but do not have permission to push to this branch:

1. Remove the unused imports (and fix the logging error) I identified above.
2. Add a readme to the task folders ([template](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/templates/new_yaml_task/README.md)), and add an entry for each to the table in `lm_eval/tasks/Readme.md`, briefly describing your benchmarks like all the other tasks. 
3. Run pre-commit:
```
pip install pre-commit
pre-commit run --all-files
```

By the way, I added `math_verify`  to `minerva_math` for equivalence checking. It uses a more general parsing regex with symbolic matching, which handles generations from more models effectively, especially for Qwen (and Qwen finetunes). Details in their [blog](https://huggingface.co/blog/math_verify_leaderboard). Just sharing for awareness!

brando90 committed:
Address PR #2946 review feedback: remove unused imports, fix logging, add READMEs, and format code

- Removed unused imports (antlr4, numpy, pandas) from utils.py and evaluator.py
- Fixed eval_logger to use logging.getLogger(__name__) instead of importing from lm_eval.utils
- Added README.md for both imo_olympiad and putnam_axiom tasks following the template
- Updated lm_eval/tasks/README.md with entries for both benchmarks
- Fixed bare except clauses to use 'except Exception:'
- Ran ruff check --fix and ruff format to ensure code formatting compliance

brando90 committed:
Add reproducible script for PR #2946 review changes

This script documents all changes made to address reviewer feedback:
- Environment setup with UV
- Code quality checks with ruff
- Summary of all changes made
- Testing instructions
- Complete workflow for future iterations

Makes it easy to reproduce the work or continue if more changes are requested.

brando90 commented:
Hi @baberabb,

Thank you for your patience and detailed feedback! I've addressed all the requested changes:

### ✅ Changes Made

1. **Removed unused imports** as identified in your review:
   - Removed `antlr4` and `numpy` from `lm_eval/tasks/imo_olympiad/utils.py`
   - Removed `antlr4` and `numpy` from `lm_eval/tasks/putnam_axiom/utils.py`
   - Removed `pandas` from `lm_eval/evaluator.py`

2. **Fixed logging** in both utils files:
   - Changed from `from lm_eval.utils import eval_logger`
   - To `eval_logger = logging.getLogger(__name__)`

3. **Added README files** following the [template](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/templates/new_yaml_task/README.md):
   - Created `lm_eval/tasks/imo_olympiad/README.md`
   - Created `lm_eval/tasks/putnam_axiom/README.md`

4. **Updated main tasks README**:
   - Added entries for both `imo_olympiad` and `putnam_axiom` to the table in `lm_eval/tasks/README.md`

5. **Code formatting**:
   - Fixed all bare `except` clauses to use `except Exception:`
   - Ran `ruff check --fix` and `ruff format` on all modified files
   - All checks now pass ✅

### 📝 Review the Changes

You can view all changes in the latest commits:
- Main changes: [`e9ae93d`](https://github.com/brando90/lm-evaluation-harness/commit/e9ae93d0)
- Reproducibility script: [`f6f18f1`](https://github.com/brando90/lm-evaluation-harness/commit/f6f18f16)

Or view the full diff on GitHub:
- [Files changed](https://github.com/EleutherAI/lm-evaluation-harness/pull/2946/files)

### 🔧 Reproducibility

I've also added a `reproduce_pr_changes.sh` script in the repo root that documents all changes and makes it easy to verify the work or make additional changes if needed.

Please let me know if there's anything else that needs to be adjusted. Thanks again for the thorough review!

Best regards,
Brando

brando90 commented:
@emxia18 does this look good to you?

brando90 commented:
@baberabb let me know what you think!

brando90 committed:
Fix formatting in README files

brando90 committed:
Address PR review feedback: fix dataset URL, remove group/task fields, remove reproduce script

brando90 commented:
@baberabb Thanks for the review! I've addressed all your feedback:
Fixed dataset URL: Updated the HuggingFace dataset URL from 5272 to 522 in the README
Removed group/task fields: Deleted the group and task fields from olympiad.yaml template
Removed reproduce script: Deleted the reproduce_pr_changes.sh file
Regarding the olympiad dataset: The olympiad tasks are currently configured to load from local JSON files (putnam-math/data/OlympiadBench_Dataset/data/OE_TO_physics_en_COMP.json). I don't see this dataset uploaded to HuggingFace yet. Should I:
Upload the olympiad dataset to HuggingFace and update the configs to use dataset_path instead of data_files?
Or keep the current local file approach for now?
The Putnam-AXIOM dataset is already on HuggingFace and working correctly. Let me know your preference for the olympiad dataset!

brando90 commented:
let me know what is needed!

brando90 commented:
@emxia18 @eric-h-chen 

Can you help me upload the imo data so we can finish this putnam PR? 

https://github.com/eric-h-chen

StellaAthena committed:
Merge branch 'main' into final_putnam_axiom_bm

emxia18 committed:
remove the imo files

brando90 commented:
@baberabb ?

baberabb commented:
Hey sorry this took so long and thanks for your patience here. Created a new PR to fix some minor nits and finally merge it.
